src/Raven.Server/Documents/PeriodicBackup/ResponsibleNodeForBackup.cs

```diff
@@ -1,102 +1,187 @@
 using System;
+using System.Collections.Generic;
+using Raven.Client.Documents.Operations.Backups;
+using Raven.Client.ServerWide;
 
 namespace Raven.Server.Documents.PeriodicBackup;
 
 public abstract class ResponsibleNodeForBackup
 {
-    protected ResponsibleNodeForBackup(string nodeTag)
+    protected ResponsibleNodeForBackup(string nodeTag, string taskName)
     {
         NodeTag = nodeTag;
+        TaskName = taskName;
     }
 
     public string NodeTag { get; }
 
+    public string TaskName { get; }
+
     public abstract ChosenNodeReason Reason { get; }
 
     public abstract string ReasonForDecisionLog { get; }
 
+    public abstract bool ShouldLog { get; }
+
     public enum ChosenNodeReason
     {
+        SameResponsibleNode,
+        SameResponsibleNodeDueToResourceLimitations,
+        SameResponsibleNodeDueToMissingHighlyAvailableTasks,
         MentorNode,
         PinnedMentorNode,
         NonExistingResponsibleNode,
         CurrentResponsibleNodeRemovedFromTopology,
-        UnchangedResponsibleNode,
         CurrentResponsibleNodeNotResponding
     }
 }
 
-public class MentorNode : ResponsibleNodeForBackup
+public abstract class ResponsibleNodeForBackupWithLimitedLogging : ResponsibleNodeForBackup
 {
-    public MentorNode(string nodeTag) : base(nodeTag)
+    private readonly long _taskId;
+    private readonly Dictionary<long, ChosenNodeReason> _lastChosenNodeReasonPerTask;
+
+    protected ResponsibleNodeForBackupWithLimitedLogging(string nodeTag, PeriodicBackupConfiguration configuration, Dictionary<long, ChosenNodeReason> lastChosenNodeReasonPerTask) : base(nodeTag, configuration.Name)
+    {
+        _taskId = configuration.TaskId;
+        _lastChosenNodeReasonPerTask = lastChosenNodeReasonPerTask;
+    }
+
+    public override bool ShouldLog
+    {
+        get
+        {
+            if (_lastChosenNodeReasonPerTask.TryGetValue(_taskId, out var chosenNodeReason)
+                && chosenNodeReason == Reason)
+            {
+                // we want to log this only once
+                return false;
+            }
+
+            return true;
+        }
+    }
+}
```

Comment:
why not to use the `LogMessage` method?

Reply:
also, you create a new instance of that every time, no?

Reply:
I need a different behavior here.
If we choose the same node twice (or more - the cluster observer is always running) for the same reason, I want to see it only once.
For example:
Mentor Node A was chosen
Mentor Node is in rehab
Mentor Node A was chosen